File: pdu/python/sensor_msgs/pdu_cdr_conv_LaserEcho.py
from .pdu_pytype_LaserEcho import LaserEcho
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_LaserEcho(writer: CdrWriter, src: LaserEcho):
    writer.write_sequence_length(src.echoes)
    for elem in src.echoes:
        writer.write_float32(elem)


def cdr_body_to_py_LaserEcho(reader: CdrReader, dst: LaserEcho):
    dst.echoes = [reader.read_float32() for _ in range(reader.read_uint32())]
    return dst


def py_to_cdr_LaserEcho(src: LaserEcho) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_LaserEcho(writer, src)
    return writer.bytes()


def cdr_to_py_LaserEcho(cdr_payload) -> LaserEcho:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = LaserEcho()
    return cdr_body_to_py_LaserEcho(reader, dst)

File: pdu/javascript/sensor_msgs/pdu_cdr_conv_LaserEcho.js
import { LaserEcho } from './pdu_jstype_LaserEcho.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduLaserEchoConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { LaserEcho } src
     */
    static to_cdr_body(writer, src) {
        writer.write_sequence_length(src.echoes);
        for (const elem of src.echoes) {
            writer.write_float32(elem);
        }
    }

    /**
     * @param {PduCdrReader} reader
     * @param { LaserEcho } dst
     * @returns { LaserEcho }
     */
    static cdr_body_to_js(reader, dst) {
        dst.echoes = Array.from({ length: reader.read_uint32() }, () => reader.read_float32());
        return dst;
    }

    /**
     * @param { LaserEcho } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { LaserEcho }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new LaserEcho());
    }
}
